app/controllers/expense_categories_controller.rb

```diff
@@ -6,6 +6,7 @@ def new
 
   def create
     @expense_category = @user.expense_categories.new(expense_category_params)
+    @expense_category.user = @user
     if @expense_category.save
       flash[:success] = ' Category created '
```

Comment:
[OPTIONAL] - I observed that even when users are not logged in, they can still reach pages that necessitate authentication. this is not good because it means your application is not secure.

To ensure that only logged-in users have privileges to perform actions or access resources, Please address this issue by restricting access to your web application resources following authorization rules. 🛠🤗

Kindly click here to learn more on [Authentication and Authorization in Ruby on Rails](https://medium.com/@zachlandis91/authentication-and-authorization-in-ruby-on-rails-c44c1ccea94d)

Reply:
As soon as you create an account you're logged in authomatically, which is what gives the illusion that an unlogged user can take action.